server/services/gameLogic.js

```diff
@@ -51,91 +83,101 @@ exports.quitGame = player => {
 }
 
 /* Adds player's card to list of played cards */
-exports.playCard = (username, card) => {
-    console.log("playCard", card);
-    if (status === statusTypes.WAITING_FOR_OTHER_PLAYERS && !allPlayers[getPlayerIndexByUsername(username)].playedCard) {
-        cardsPlayed.push({id: card, hidden: true});
-        allPlayers[getPlayerIndexByUsername(username)].playedCard = true;
+exports.playCard = (username, cardId) => {
+    if (status === statusTypes.WAITING_FOR_OTHER_PLAYERS && !playerHasPlayedCard(username)) {
+        const card = {
+            username: username,
+            cardId: cardId
+        };
+        cards.push(card);
         if (allPlayersPlayedCard()) {
             status = statusTypes.WAITING_FOR_VOTES;
-            socket.emitStatus(status);
-            socket.emitPlayedCards(cardsPlayed);
+            socket.emitPlayedCards(getCards());
             socket.promptPlayersVote(currentPlayer);
         }
     }
 }
 
 /* Vote for a card */
-exports.voteCard = (username, card) => {
-    // console.log("voteCard/end turn");
-    if (status === statusTypes.WAITING_FOR_VOTES && !allPlayers[getPlayerIndexByUsername(username)].voted) {
-        votes.push({id: card, user: username});
-        allPlayers[getPlayerIndexByUsername(username)].voted = true;
-        if (allPlayersVoted()) {
-            incrementRound();
-        }
+exports.voteCard = (username, cardId) => {
+    if (status === statusTypes.WAITING_FOR_VOTES && !playerHasVoted(username)) {
+        const vote = {
+            username: username,
+            cardId: cardId
+        };
+        votes.push(vote);
+        if (allPlayersVoted()) nextRound();
     }
 }
 
-/* Get the index of the player in the list */
-const getPlayerIndexByUsername = username => {
-    return allPlayers.findIndex(player => player.username === username);
+/* Return true if the player has already played this round */
+const playerHasPlayedCard = (username) => {
+    return cards.filter(card => card.username === username).length > 0;
 }
 
-/* Move on to the next round, called when all players have finished their turn */
-const incrementRound = () => {
-    // console.log("Everyone done, next round!");
-    roundNum++;
-    allPlayers = allPlayers.map(player => {
-        return {...player, playedCard: false, voted: false};
-    });
-    currentPlayer = allPlayers[roundNum % allPlayers.length];
-    cardsPlayed = [];
-    currentWord = '';
-    socket.promptCurrentPlayer(currentPlayer);
+/* Return true if the player has already voted this round */
+const playerHasVoted = (username) => {
+    return votes.filter(vote => vote.username === username).length > 0;
 }
 
 /* Returns true if all players have played a card this round */
 const allPlayersPlayedCard = () => {
-    for (let player of allPlayers.filter(player => currentPlayer !== player)) {
-        if (!player.playedCard) return false;
+    for (let player of players) {
+        if (!playerHasPlayedCard(player.username)) return false;
```

Comment:
These two `for` loops in `allPlayersPlayedCard` and `allPlayersVoted` could be implemented using `every`.
`const allPlayersPlayedCard = () => players.every(player => playerHasPlayedCard(player))`
Something like that.